cli/src/cmd/cast/storage.rs

```diff
@@ -0,0 +1,206 @@
+            NameOrAddress::Name(name) => provider.resolve_name(&name).await?,
+            NameOrAddress::Address(address) => address,
+        };
+
+        // Slot was provided, perform a simple RPC call
+        if let Some(slot) = slot {
+            let cast = Cast::new(provider);
+            println!("{}", cast.storage(address, slot, block).await?);
+            return Ok(())
+        }
+
+        // No slot was provided
+
+        // Get deployed bytecode at given address
+        let address_code = provider.get_code(address, block).await?;
+        if address_code.is_empty() {
+            eyre::bail!("Provided address has no deployed code and thus no storage");
+        }
+
+        // Check if we're in a forge project
+        let project = build.project()?;
+        if project.paths.has_input_files() {
+            // Find in artifacts and pretty print
+            let project = with_storage_layout_output(project);
+            let out = compile(&project, false, false)?;
+            let artifact = out.artifacts().find(|(_, artifact)| match artifact.deployed_bytecode {
+                Some(ref deployed_code) => match deployed_code.bytecode {
+                    Some(ref bytecode) => match bytecode.object.as_bytes() {
+                        Some(bytes) => bytes == &address_code,
+                        None => false,
+                    },
+                    None => false,
+                },
+                None => false,
+            });
+            if let Some((_, artifact)) = artifact {
+                return print_storage_layout(&artifact.storage_layout, true)
+            }
+        }
+
+        // Not a forge project or artifact not found
+        // Get code from Etherscan
+        let api_key = etherscan_api_key.or_else(|| {
+            let config = Config::load();
+            config.get_etherscan_api_key(Some(chain))
+        }).ok_or_else(|| eyre::eyre!("No Etherscan API Key is set. Consider using the ETHERSCAN_API_KEY env var, or setting the -e CLI argument or etherscan-api-key in foundry.toml"))?;
+        let client = ethers::etherscan::Client::new(chain, api_key)?;
+        println!("No artifacts found, fetching source code from etherscan...");
+        let source = client.contract_source_code(address).await?;
+        let source_tree = source.source_tree()?;
+
+        // Create a new temp project
+        // let root = tempfile::tempdir()?;
+        let root = std::path::PathBuf::from(concat!(env!("CARGO_MANIFEST_DIR"), "/temp_build"));
+        println!("root: {}", root.display());
+        std::fs::create_dir_all(&root)?;
+        source_tree.write_to(&root)?;
+
+        // Configure Solc
+        let paths = ProjectPathsConfig::builder().sources(&root).build_with_root(root);
+
+        let metadata = &source.items[0];
+        let mut settings = Settings::default();
+
+        let mut optimizer = Optimizer::default();
+        if parse_etherscan_bool(&metadata.optimization_used)? {
```

Comment:
imo this seems unnecessary, I think we can rely on etherscan sending only 0 or 1 here since this is a bool.